state-chain/pallets/cf-swapping/src/migrations/swapping_redesign.rs

```diff
@@ -0,0 +1,96 @@
+use crate::*;
+use core::marker::PhantomData;
+use frame_support::traits::OnRuntimeUpgrade;
+
+pub struct Migration<T: Config>(PhantomData<T>);
+
+mod old {
+
+	use super::*;
+
+	#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo)]
+	pub struct CcmSwap {
+		source_asset: Asset,
+		deposit_amount: AssetAmount,
+		destination_asset: Asset,
+		destination_address: ForeignChainAddress,
+		deposit_metadata: CcmDepositMetadata,
+		principal_swap_id: Option<SwapId>,
+		gas_swap_id: Option<SwapId>,
+	}
+
+	#[derive(
+		Clone, Copy, Debug, Default, PartialEq, Eq, Encode, Decode, TypeInfo, MaxEncodedLen,
+	)]
+	pub struct CcmSwapOutput {
+		principal: Option<AssetAmount>,
+		gas: Option<AssetAmount>,
+	}
+
+	#[frame_support::storage_alias]
+	pub type CcmIdCounter<T: Config> = StorageValue<Pallet<T>, u64, ValueQuery>;
+
+	#[frame_support::storage_alias]
+	pub type PendingCcms<T: Config> = StorageMap<Pallet<T>, Twox64Concat, u64, CcmSwap>;
+
+	#[frame_support::storage_alias]
+	pub type CcmOutputs<T: Config> = StorageMap<Pallet<T>, Twox64Concat, u64, CcmSwapOutput>;
+}
+
+impl<T: Config> OnRuntimeUpgrade for Migration<T> {
+	fn on_runtime_upgrade() -> Weight {
+		// Note: we don't migrate items from SwapQueue because we will
+		// ensure that it is empty during the upgrade.
+
+		log::info!("Swapping redesign migration!");
+
+		let latest_used_ccm_id = old::CcmIdCounter::<T>::get();
+		let latest_used_swap_id = SwapIdCounter::<T>::get();
+
+		SwapRequestIdCounter::<T>::put(latest_used_ccm_id + latest_used_swap_id);
```

Comment:
i think from the product side it would be okay to only use `latest_used_swap_id` here, but passing the sum of them should be fine too (maybe im missing something though @acdibble)

Reply:
I thought in our discussion we decided to reserve ids in the range `(latest_used_swap_id, latest_used_swap_id + latest_used_ccm_id]` for existing CCM swaps (since requets ids will be replacing ccm_ids)?